### ISSUE TITLE
Background download of Firefox Browser failed

### ISSUE THREAD

**Issue by Chefski** [Tobi823/ffupdater#206]
Stacktrace:
```
de.marmaro.krt.ffupdater.app.impl.exceptions.NetworkException: File download failed.
	at de.marmaro.krt.ffupdater.download.FileDownloader$downloadFileAsync$2$1.invokeSuspend(Unknown Source:90)
	at t3.a.resumeWith(Unknown Source:8)
	at g4.d0.run(Unknown Source:94)
	at j4.a$a.run(SourceFile:9)
Caused by: java.net.SocketTimeoutException: timeout
	at s4.q$c.l(SourceFile:1)
	at s4.q$b.read(SourceFile:20)
	at p4.c$b.read(Unknown Source:15)
	at x4.s.read(SourceFile:4)
	at x4.j.read(Unknown Source:7)
	at de.marmaro.krt.ffupdater.download.ProgressResponseBody$trackTransmittedBytes$1.read(Unknown Source:5)
	at x4.s$a.read(SourceFile:4)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:347)
	at java.io.FilterInputStream.read(FilterInputStream.java:107)
	at v2.e.L(Unknown Source:23)
	at de.marmaro.krt.ffupdater.download.FileDownloader.downloadFileInternal(SourceFile:3)
	at de.marmaro.krt.ffupdater.download.FileDownloader.access$downloadFileInternal(Unknown Source:0)
	at de.marmaro.krt.ffupdater.download.FileDownloader$downloadFileInternal$1.invokeSuspend(Unknown Source:12)
	... 3 more
```
Device information:
| key | value |
| --- | --- |
| version name | 76.0.1 |
| version code | 107 |
| build type | release |
| SDK version | 31 |
| android release | 12 |
| board | kona |
| brand | OnePlus |
| device | OnePlus8Pro |
| manufacturer | OnePlus |
| model | IN2023 |
| product | OnePlus8Pro |
| SOC manufacturer | Qualcomm |
| SOC model | SM8250 |
| supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| tags | release-keys |
| build time | 1654882028000 |

**Comment by Tobi823** [comment]
> If it's a known issue then it's fine. I was just reporting to make sure everything is fine

Thanks, I was not aware that in rare cases FFUpdater cannot successfully check for updates within 48 hours due to network issues.

(FYI: If the background update check fails, it will be restarted with an increasing delay: after first failure 15s, after second failure 30s, after third failure 1m, 2m, 4m, 8m, 16m, 32m, 64m, 128m, 256m, 300m, 300m, 300m, 300m)

(In hindsight - even I leave my tablet in flight mode for 2-3 days when I don't need it.)

I think the best solution is:
- show warning after 5 days without a successful update check
- show a different notification. Something like "In the last 5 days FFUpdater was not able to check for updates in the background. If you had your smartphone in flight mode for a long time, you can ignore this message. If not, click here to view the error report."

**Comment by Tobi823** [comment]
I will publish the fix in version 77.0.2 on Friday.

**Comment by Tobi823** [comment]
77.0.3 is now published
